src/java/org/apache/cassandra/utils/obs/MemoryLimiter.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils.obs;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.apache.cassandra.utils.FBUtilities;
+
+public class MemoryLimiter
+{
+    public final long maxMemory;
+    private final AtomicLong currentMemory;
+    private final String exceptionFormat;
+
+    public MemoryLimiter(long maxMemory, String exceptionFormat)
+    {
+        this.maxMemory = maxMemory;
+        this.currentMemory = new AtomicLong();
+        this.exceptionFormat = exceptionFormat;
```

Comment:
I know it looks so it could be, but I don't want to mix concepts - why a distinct `MemoryLimiter` component should have a fixed exception format saying it is all about a bloom filter - this is one of the reasons I didn't get the solution directly from DSE and refactored this stuff

Reply:
I see, make sense 👍🏻